pkgs/applications/audio/friture/unlock_constraints.patch

```diff
@@ -1,34 +1,30 @@
 diff --git a/setup.py b/setup.py
-index 4092388..6cb7dac 100644
+index 886fe78..5dccac0 100644
 --- a/setup.py
 +++ b/setup.py
-@@ -50,19 +50,19 @@ ext_modules = [LateIncludeExtension("friture_extensions.exp_smoothing_conv",
+@@ -50,17 +50,17 @@ ext_modules = [LateIncludeExtension("friture_extensions.exp_smoothing_conv",
  # these will be installed when calling 'pip install friture'
  # they are also retrieved by 'requirements.txt'
  install_requires = [
 -    "sounddevice==0.4.2",
 -    "rtmixer==0.1.3",
--    "PyOpenGL==3.1.5",
--    "PyOpenGL-accelerate==3.1.5",
 -    "docutils==0.17.1",
 -    "numpy==1.21.1",
 -    "PyQt5==5.15.4",
 -    "appdirs==1.4.4",
 -    "pyrr==0.10.3",
-+    "sounddevice>=0.4.1",
-+    "rtmixer>=0.1.1",
-+    "PyOpenGL>=3.1.4",
-+    "PyOpenGL-accelerate>=3.1.5",
-+    "docutils>=0.17.1",
-+    "numpy>=1.20.3",
-+    "PyQt5>=5.15.4",
-+    "appdirs>=1.4.4",
-+    "pyrr>=0.10.3",
++    "sounddevice<=0.4.2",
++    "rtmixer<=0.1.3",
++    "docutils<=0.17.1",
++    "numpy>=1.21.1",
++    "PyQt5<=5.15.4",
++    "appdirs<=1.4.4",
++    "pyrr<=0.10.3",
```

Comment:
Just substituteReplace ``==`` with ``>=``.

Reply:
To expand a bit more on this.
nixpkgs has a function called `substituteInPlace` in stdenv which is a function to replace a string in a file.
if you grep into nixpkgs you should find many uses of it.
When I first saw this, I had a question why do we use a custom function rather than sed for example. Basically that function has additional checks and warnings. For example it will fail if you try to apply it on a binary file. It will warn you if the pattern you are trying to match, doesn't actually match.
Feel free to ask questions if anything isn't clear.

Reply:
Hi, thanks for taking the time to explain this. However, this is very much intentional - numpy is 1.21.1 in Nixpkgs and needs `>=` in the patch, while the others do in fact need `<=`. It might be a better idea to remove the version requirements altogether though.

Reply:
I've now changed the patch into an `sed` command.